pkg/kubectl/cmd/run.go

```diff
@@ -425,6 +426,15 @@ func createGeneratedObject(f *cmdutil.Factory, cmd *cobra.Command, generator kub
 		return nil, "", nil, nil, err
 	}
 
+	annotations, err := mapping.MetadataAccessor.Annotations(obj)
+	if err != nil {
+		return nil, "", nil, nil, err
+	}
+	if cmdutil.GetRecordFlag(cmd) || len(annotations[kubectl.ChangeCauseAnnotation]) > 0 {
```

Comment:
`RecordChangeCause` just updates the runtime.Object by adding change-cause annotation to it. In L446 we get the info from this object, and in L455 we create the object with resource.NewHelper.Create.